sdk/textanalytics/azure-ai-textanalytics/azure/ai/textanalytics/_request_handlers.py

```diff
@@ -32,6 +32,9 @@ def _validate_batch_input(documents, hint, whole_batch_hint):
         if not all(isinstance(x, (dict, TextDocumentInput, DetectLanguageInput)) for x in documents):
             raise TypeError("Mixing string and dictionary/object document input unsupported.")
 
+    if not isinstance(documents, list):
+        raise TypeError("`documents` parameter must be a list, not a {}".format(type(documents)))
```

Comment:
we shouldn't do this type of validation - a user should be able to pass any iterable

Reply:
Ok, I see what you're saying. I added this because I accidentally passed a dict instead of a dict in a list ({"id": 1, "text": "hello", "language": "es"}), and it went through, but passed in three documents to the generated code (one for each entry in the dict).

I'll at least add the check to make sure documents isn't a dict (since that's an iterable). Do you think there's anything else I should check?